### ISSUE TITLE
Cli disablement options.

### ISSUE THREAD

**Issue by bend-n** [godotengine/godot-proposals#4815]
### Describe the project you are working on

A cli parser addon for my projects.

### Describe the problem or limitation you are having in your project

```
-h
-v
-e
-p
-q
-l
-u
-f
-m
-w
-t
-d
-b
-s
```
Are all taken, as CLI arguments. If you do, say, `godot -f` godot gulps the argument, and your `OS.get_cmdline_args()` returns `[]`.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I am now able to override `-h`, `-v`, `-e`, `-p`, etc 🥳!

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

 I think there should at *least* be a project setting that disables godot from gulping the arguments, at medium a option that completely turns off godot's CLI handling, and at best a option/method that lets **YOU** handle the arguments **FIRST** and then pass the leftovers to godot.

So for turning off gulping, one of these little tickboxes ![image](https://user-images.githubusercontent.com/70787919/177282079-37fc4fb0-c0c2-4b91-9211-95ca0fff1c02.png) in project settings should be sufficient
ditto for disabling godot's CLI handling,
And for the last option, something like 
```
var args := OS.get_cmdline_args()
args.remove(len(args.size))
OS.handle_args(args)
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not really; You can force use of `--arg=stuff,2stuff,3stuff`, but its unintuitive, and you can create a wrapper script that turns `--arg stuff stuff2 stuff3` into `--arg=stuff,stuff2,stuff3`, but it will need to be generated with code if you want it to only take 3 arguments: `--arg stuff stuff UNRELATED STUFF` will be turned into `--arg=stuff,stuff,UNRELATED,STUFF` if its just a simple `sed` script, and if you want to multi-platform your game, then you have to maintain different wrapper scripts per platform.

### Is there a reason why this should be core and not an add-on in the asset library?

Not possible

**Comment by bend-n** [comment]
https://github.com/godotengine/godot-proposals/issues/2797 is *somewhat* related, but still wants the engine to have it's own args.

**Comment by YuriSizov** [comment]
> and at best a option/method that lets YOU handle the arguments FIRST and then pass the leftovers to godot.

I mean, that's impossible. Godot needs to parse and consume command line arguments to launch your project in the first place. Some of the options define the work environment for the runtime too, and probably can't be set up afterwards.

Either way, I think this proposal is missing both technical know-how and a good explanation of the use case.

**Comment by bend-n** [comment]
> I mean, that's impossible. Godot needs to parse and consume command line arguments to launch your project in the first place. Some of the options define the work environment for the runtime too, and probably can't be set up afterwards.

Objection! The ***exported*** godot needs not to handle cli arguments.

**Comment by Calinou** [comment]
> The exported godot needs not to handle cli arguments.

A lot of people require their exported projects to handle command line arguments. This is the case for non-game applications using Godot, but also many Godot games exposing "power user" options.

**Comment by YuriSizov** [comment]
@Calinou They are saying that the Godot executable of an exported project doesn't need to handle our default arguments (`-f`, `-e`, `--path`, etc).

**Comment by bend-n** [comment]
> > The exported godot needs not to handle cli arguments.
> 
> A lot of people require their exported projects to handle command line arguments. This is the case for non-game applications using Godot, but also many Godot games exposing "power user" options.

Assuming you mean the `--fullscreen` options and such, thats why i think it would work if you handled the args, handle it yourself, and then pass it to godot yourself, OR implement custom `--fullscreen` options, which would be the way to go if all cli handling was disabled.

**Comment by Calinou** [comment]
> @Calinou They are saying that the Godot executable of an exported project doesn't need to handle our default arguments (`-f`, `-e`, `--path`, etc).

I don't like the direction this is going in, as many CLI arguments are useful to power users in exported projects. Not all games will implement extensive settings menus (especially gamejam games), but sometimes having *some way* to override a parameter makes the difference between an unplayable and a playable game to some.

**Comment by bend-n** [comment]
I mean of course the optimal way to do this (but also the hardest) would be to have some kind of user cli script, and to have a cli arg handler class that the cli script can touch

**Comment by bend-n** [comment]
Also, could just add a `--script` arg that lets you put in a `Window.fullscreen = true`, then you can have full powa.

**Comment by Calinou** [comment]
> Also, could just add a `--script` arg that lets you put in a `Window.fullscreen = true`, then you can have full powa.

This still makes usability more difficult, as you have to write a script, save it to disk, possibly learn GDScript, etc. Remember that players are generally not Godot users.

**Comment by bend-n** [comment]
Your saying that theyre power users, not random players that dont know how to use godot.

**Comment by Calinou** [comment]
> Your saying that theyre power users, not random players that dont know how to use godot.

What I mean is that "power users" of a game are generally not Godot users. They just play random games and don't care (too much) about the engine it's running on – until it makes bad decisions :slightly_smiling_face:

**Comment by bend-n** [comment]
I would still like to see a game that gets broken by this; And also, how did they learn the arguments anyways?
If they learnt the arguments from the documentation, we could just put in a page that details how to use it.
At the very least, I want to remove the short form arguments, as they are unreadable, and have conflicts with your cli.

**Comment by Zireael07** [comment]
> Any change of this kind needs to be strictly opt-in

Agreed. @bend-n Consider that godot uses CLI arguments to specify renderer, or to run the project in current folder.

**Comment by bend-n** [comment]
Ive addressed those 2 examples:
renderer argument will stay, and editor builds will have all args.
